profiler/profiler.go

```diff
@@ -22,25 +22,23 @@ const outChannelSize = 5
 var (
 	mu             sync.Mutex
 	activeProfiler *profiler
+	containerID    = internal.ContainerID()
 )
 
-// ErrMissingAPIKey is returned when an API key was not found by the profiler.
-var ErrMissingAPIKey = errors.New("API key is missing; provide it using the profiler.WithAPIKey option")
-
 // Start starts the profiler. It may return an error if an API key is not provided by means of
 // the WithAPIKey option, or if a hostname is not found.
 func Start(opts ...Option) error {
 	cfg := defaultConfig()
 	for _, opt := range opts {
 		opt(cfg)
 	}
-	if cfg.apiKey == "" {
```

Comment:
If the user legitimately forgets to add his API key, how will he know now? We are making odd assumptions here. I tend to think it would make sense to add an option for this, maybe `DD_PROFILER_USE_AGENT` and `profiler.WithAgentProxy` or something...

Otherwise it just seems a bit forced. Not a huge fan of this assumption that if API key is empty we use the agent.

Reply:
This is precisely what we want. The old method of providing your own API key at app-level will be, at least initially, deprecated and removed from setup docs. It's only supported so existing users won't have their current setups broken with an update of dd-trace-go.

Reply:
Got it. Resolving.